handler.go

```diff
@@ -123,17 +123,28 @@ func (fm *Frontman) PostResultsToHub(results []Result) error {
 		}
 	}
 
+	fm.offlineResultsBuffer = append(fm.offlineResultsBuffer, results...)
 	// Bundle hostInfo and results
-	resultsToSend := Results{
+	b, err := json.Marshal(Results{
+		Results: fm.offlineResultsBuffer,
 		HostInfo: hostInfo,
-		Results:  results,
-	}
+	})
 
-	b, err := json.Marshal(resultsToSend)
 	if err != nil {
 		return err
 	}
 
+	// in case buffer + results exceed HubMaxOfflineBufferBytes, reset buffer
+	if len(b) > fm.HubMaxOfflineBufferBytes && len(fm.offlineResultsBuffer) > 0 {
+		log.Errorf("hub_max_offline_buffer_bytes(%d bytes) exceed with %d results. Flushing the buffer...", fm.HubMaxOfflineBufferBytes, len(results))
+
+		fm.offlineResultsBuffer = []Result{}
```

Comment:
Should we discard _the whole_ buffer here or maybe just enough to fit `hub_max_offline_buffer_bytes `?

Reply:
it is pretty tricky. Because
- we measure the JSON-marshaled struct
- there is no way to get the size of struct directly without `unsafe`

Reply:
I guess we would have to repeat removing an item from `Result` and marshalling again until the size fits. If you say it's fine to simply clean the whole buffer I'm fine with that.